src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -21,48 +25,74 @@
 @WebServlet("/matches")
 public class MatchServlet extends HttpServlet {
 
+  private UserService userService;
+  private NonPersistentUserRepository userRepository;
+
   @Override
   public void init(ServletConfig config) throws ServletException {
     super.init(config);
+    userRepository = new NonPersistentUserRepository();
+    MatchRepository matchRepository = new NonPersistentMatchRepository();
+
+    getServletContext().setAttribute("matchRepository", matchRepository);
+    setLoggedInUser();
+  }
 
-    NonPersistentMatchRepository repository = new NonPersistentMatchRepository();
-    User testUser = repository.addTestData();
+  public void setLoggedInUser() {
+    User currentUser;
+    
+    userService = UserServiceFactory.getUserService();
+    com.google.appengine.api.users.User currentGoogleUser = userService.getCurrentUser();
+    if(currentGoogleUser == null) {
+      currentUser = null;
+    } else {
+      currentUser = userRepository.getUser(currentGoogleUser);
+    }
 
-    //Set MatchRepository and Current User as ServletContext so that they can be accessed by all servlets
-    getServletContext().setAttribute("matchRepository", repository);
-    getServletContext().setAttribute("currentUser", testUser);
+    getServletContext().setAttribute("currentUser", currentUser);
```

Comment:
One problem with the current implementation is currentUser will be set to the last User who sent a GET request while the POST requests use currentUser. 

If we have multiple users accessing the site simultaneously, then User A could save matches to User B's list:

1. User A -> GET /matches (currentUser = User A): User A accesses matches page
2. User B -> GET /matches (currentUser = User B): User B accesses matches page
3. User A -> POST /new-matches-servlet (currentUser = User B): User A request matches
4. User A -> POST /matches (currentUser = UserB): User A saves matches for User B 

Treating each GET and POST request as independent requests may be helpful here. The Google User API should allow you to determine the user state for each request rather than saving it in the servlet context.

Reply:
I took out the User ServletContext and added a method to get the current user which I call in each method. Let me know if this is what you meant, thanks!